code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -132,11 +115,21 @@
 	if(combined_heat_capacity > 0)
 		var/combined_energy = heat_capacity * target_temperature + air_heat_capacity * air_contents.temperature
 		air_contents.temperature = combined_energy/combined_heat_capacity
+		if(cooling ^ old_temperature > target_temperature)
```

Comment:
The first time i saw this, it wasn't quite clear to me what you were doing, as it looked like you were XORing cooling and old_temperature together before checking againt target_temperature, this makes it clear what you're trying to do
```suggestion
		if(cooling ^ (old_temperature > target_temperature))
```